noir-projects/noir-protocol-circuits/crates/types/src/constants.nr

```diff
@@ -59,7 +55,9 @@ global NUM_UNENCRYPTED_LOGS_HASHES_PER_TX: u64 = 1;
 // docs:end:constants
 
 // KERNEL CIRCUIT PRIVATE INPUTS CONSTANTS
-global MAX_PUBLIC_DATA_HINTS: u64 = 64; // MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX + MAX_PUBLIC_DATA_READS_PER_TX;
+// global MAX_PUBLIC_DATA_HINTS: u64 = MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX + MAX_PUBLIC_DATA_READS_PER_TX;
+// FIX: Sadly, writing this as above causes a type error in type_conversion.ts.
+global MAX_PUBLIC_DATA_HINTS: u64 = 64;
```

Comment:
Is this because ts is not happy with building tuples of calculated lengths? Maybe we could evaluate the arithmetic expressions when converting into ts then?

Reply:
We could! Step by step :)